+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -775,8 +779,145 @@ function create_solver(obj, solver_options, plugin, regenerate)
+        end
+
+        function [IG,IH,I00] = process_active_set(obj, active_set)
+        % This method takes a nosnoc active set for a PSS and produces an active set for the
+        % complementarity constraints in this problem. It returns these as a boolean array.
+        %
+        % TODO(@anton) there should be a cleaner way of doing this but this is "easy".
+        % TODO(@anton) this assumes no user complementarities, but how do we handle those?
+        % TODO(@anton) This only works if we lift c. in principle this is why we should backcalculate.
+            arguments
+                obj
+                active_set nosnoc.activeset.Pds
+            end
+            dims = obj.dcs.dims;
+            dcs = obj.dcs;
+            model = obj.model;
+            opts = obj.opts;
+
+            % TODO(@anton) handle initial algebraics
+            active_constraint_0 = active_set.active_constraints{1};
+
+            c_values = ones(dims.n_c,1);
+            lambda_values = zeros(dims.n_c,1);
+            if ~isempty(active_constraint_0)
+                c_values(active_constraint_0) = 0;
+                lambda_values(active_constraint_0) = 1;
+            end
+            obj.w.c_lift(0,0,opts.n_s).init = c_values;
+            obj.w.lambda(0,0,opts.n_s).init = lambda_values;
+
+            if ~isempty(active_set.times)
+                jj = 1; % Control stage index
+                kk = 0; % FE index
+                t_curr = 0;
+                % Go through the active set by times
+                for ii=1:active_set.get_n_steps()
+                    done_stage = false;
+                    t_end = active_set.times(ii);
+
+                    active_constraint_ii = active_set.active_constraints{ii};
+                    n_active = length(active_constraint_ii);
+
+                    c_values = ones(dims.n_c,1);
+                    lambda_values = zeros(dims.n_c,1);
+                    if ~isempty(active_constraint_ii)
+                        c_values(active_constraint_ii) = 0;
+                        lambda_values(active_constraint_ii) = 1;
+                    end
+                    for jj=jj:opts.N_stages
+                        for kk=(kk+1):opts.N_finite_elements(jj)
+                            for ll=1:opts.n_s
+                                obj.w.c_lift(jj,kk,ll).init = c_values;
+                                obj.w.lambda(jj,kk,ll).init = lambda_values;
+                            end
+                            t_curr = t_curr + opts.h_k(jj);
+                            if t_curr > t_end
+                                done_stage = true;
+                                break
+                            end
+                        end
+                        if done_stage
+                            break
+                        end
+                        kk = 0; % Reset fe counter
+                    end
+                    % handle entering active_constraints TODO(@anton) verify this is correct
```

Comment:
I guess the purpose if this is to account for cross comp of adjecent finite elements. (if yes comment so).

I believe this is correctly implemented, see 
examples/pass_pss_active_set/simulation_pass_active_set.m 
whit nice plots of lambda and theta, everything seems fine.